odxtools/parameters/parameterwithdop.py

```diff
@@ -17,22 +18,13 @@
     from ..diaglayer import DiagLayer
 
 
+@dataclass
 class ParameterWithDOP(Parameter):
+    dop_ref: Optional[OdxLinkRef]
+    dop_snref: Optional[str]
 
-    def __init__(
-        self,
-        *,
-        parameter_type: str,
-        dop_ref: Optional[OdxLinkRef],
-        dop_snref: Optional[str],
-        **kwargs,
-    ) -> None:
-        super().__init__(parameter_type=parameter_type, **kwargs)
-        self.dop_ref = dop_ref
-        self.dop_snref = dop_snref
-
-        self._dop: Optional[DopBase] = None
-        if dop_snref is None and dop_ref is None:
+    def __post_init__(self) -> None:
+        if self.dop_snref is None and self.dop_ref is None:
             logger.warn(f"Param {self.short_name} without DOP-(SN)REF should not exist!")
```

Comment:
Why not use warnings module?

Reply:
The warning should indicate to what parameter it is referring

Reply:
historical artifact. instead of `warnings`, this should use the use the standard odxtools error handling mechanism, though (i.e., `odxassert()`). fixed: b282cb8f7c1ac63e6635f139fbab9281b0a1add8